OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -176,6 +189,18 @@ public void Repulse()
 			SetPosition(self, CenterPosition + FlyStep(speed, repulsionForce.Yaw.Facing));
 		}
 
+		public void DisableRepulsing()
```

Comment:
There is already a check in GetRepulsionForce that disables the repulsion if the actor isn't at its cruising altitude.  I suspect the problem is that it gets repulsed during the one tick where it is at cruising height and preparing to land.  You could change that check to additionally disable repulsion if the actor is within X WRange of the reserved docking point.